### ISSUE TITLE
[in_app_purchase] There's no way - other than completePurchase - to remove/clear a purchase for which verification failed 

### ISSUE THREAD

**Issue by jesper372** [flutter/flutter#110665]
From the plugin README:

> After verifying the purchase receipt and the delivering the content to the user it is important to call InAppPurchase.completePurchase to tell the underlying store that the purchase has been completed. Calling InAppPurchase.completePurchase will inform the underlying store that the app verified and processed the purchase and the store can proceed to finalize the transaction and bill the end user's payment account.

Now, if the verification of the purchase receipt failed, e.g. due to a connection error when attempting to contact the backend that does the verification, then as implied by the above, `completePurchase` should not be called, as the underlying store will then proceed to finalize the transaction and bill the end user's payment account, even though the purchase hasn't been verified and processed.

But this leaves the question of how to remove/clear the failed purchase, given that `completePurchase` appears to be the only way to do so. Not clearing it leads to the following exception if the user makes another attempt at purchasing the product:

> There is a pending transaction for the same product identifier. Please either wait for it to be finished or finish it manually using `completePurchase` to avoid edge cases.

In other words, there are seemingly only two options for how to deal with a failed purchase:
1) Don't call `completePurchase` purchase on it, in which case the user is unable to reattempt making the purchase, or
2) Do call `completePurchase` purchase on it, in which case the underlying store is erroneously informed that the purchase has been successfully verified and processed.

Neither option makes sense from a usability standpoint; is this correct, or am I missing something?

**Comment by darshankawar** [comment]
Thanks for the report @jesper372 
Can you provide on which platform are you seeing this behavior ? (Android or iOS )

**Comment by jesper372** [comment]
@darshankawar Apologies; I'm closing this, because I now understand that the logic underlying the behaviour described above is that once a purchase has been cleared by the underlying store, the verification of the purchase really must succeed; if not immediately (e.g. because the backend cannot be contacted), then at least subsequently. Thus, the purchase should remain in the queue until the verification does get done, and completePurchase has been called on it.
